src/cbapi/psc/threathunter/query.py

```diff
@@ -254,21 +254,24 @@ def not_(self, q=None, **kwargs):
 
     def _get_query_parameters(self):
         args = self._default_args.copy()
-        args['q'] = self._query_builder._collapse()
+        args['query'] = self._query_builder._collapse()
+
+        # TODO move this cb.process_guid somewhere
         if self._query_builder._process_guid is not None:
             args["cb.process_guid"] = self._query_builder._process_guid
-        args["fl"] = "*,parent_hash,parent_name,process_cmdline,backend_timestamp,device_external_ip,device_group," \
-            + "device_internal_ip,device_os,process_effective_reputation,process_reputation,ttp"
+        args["fields"] = ["*","parent_hash","parent_name","process_cmdline","backend_timestamp","device_external_ip",
```

Comment:
Isn't the switch from "fl" to "fields" only possible if you switch from the v1 to the v2 route?  (or is this one of the reasons why the PR is marked WIP?). I see it updated to v2 in some of routes here in `query.py` but not in any of the routes in `models.py` or `rest_api.py` and wasn't sure if that's a conflict or if it's expected.

Reply:
You're right. That's part of the reason it's WIP. I know the code as it sits is not 100% done and I'm still unfamiliar with a lot of this code. While this PR is a start in the right direction, I think I still need a bit of help getting it across the finish line.